src/edu/ucsd/library/xdre/web/CollectionStatusReportController.java

```diff
@@ -0,0 +1,260 @@
+package edu.ucsd.library.xdre.web;
+
+import java.io.OutputStream;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.log4j.Logger;
+import org.dom4j.Document;
+import org.dom4j.Node;
+import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.servlet.mvc.Controller;
+
+import edu.ucsd.library.xdre.statistic.analyzer.Statistics;
+import edu.ucsd.library.xdre.utils.Constants;
+import edu.ucsd.library.xdre.utils.DAMSClient;
+
+/**
+ * Class CollectionsReportController generate reports for DAMS collection changes
+ * @author lsitu
+ *
+ */
+public class CollectionStatusReportController implements Controller {
+    private static Logger log = Logger.getLogger(CollectionStatusReportController.class);
+    private static SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
+
+    public ModelAndView handleRequest(HttpServletRequest request, HttpServletResponse response) throws Exception {
+        String beginDate = request.getParameter("beginDate");
+        String endDate = request.getParameter("endDate");
+        boolean exportOnly = request.getParameter("export") != null;
+
+        DAMSClient damsClient = null;
+        DAMSClient damsClientEvent = null; // query events
+        Map<String, Object> dataMap = new HashMap<>();
+        if (StringUtils.isNotBlank(beginDate)) {
```

Comment:
Can `beginDate` be blank at this point after your client-side validation? And I guess, if it can, should it check for `endDate` too?